forms/account-freemarker/src/main/java/org/keycloak/account/freemarker/FreeMarkerAccountProvider.java

```diff
@@ -197,7 +197,11 @@ public Response createResponse(AccountPages page) {
             String result = freeMarker.processTemplate(attributes, Templates.getTemplate(page), theme);
             Response.ResponseBuilder builder = Response.status(status).type(MediaType.TEXT_HTML).entity(result);
             BrowserSecurityHeaderSetup.headers(builder, realm);
-            LocaleHelper.updateLocaleCookie(builder, locale, realm, uriInfo, Urls.localeCookiePath(baseUri,realm.getName()));
+
+            String keycloakLocaleCookiePath = Urls.localeCookiePath(baseUri, realm.getName());
```

Comment:
I think you're probably right.  But I don't see the selected locale in UserModel.  Is it currently there?

I think if i18n is enabled for a realm then the selected locale should be automatically added to the token.  The rationale being, if you say you want to use i18n then there is no reason you wouldn't want to expose preferred locale.

Shouldn't we get locale from the token as well?